### PR TITLE
CODE-2807 Split CampaignSourceEntry

### PR DESCRIPTION
**Pull request by thpr**

### CODE REVIEW

**Comment by LegacyKing** [comment]
@jdempsey-au flagging for your review.

File: code/src/java/pcgen/persistence/lst/CampaignSourceEntry.java [hunk 1]
```diff
@@ -97,65 +74,22 @@ public CampaignSourceEntry(Campaign campaign, URI lstLoc)
...
 	{
 		super();
 		if (campaign == null)
 		{
 			throw new IllegalArgumentException("campaign can't be null");
 		}
-		if (fac == null)
+		if (entry == null)
 		{
 			throw new IllegalArgumentException("URI Factory can't be null");
```
Comment (jdempsey-au) [inline code/src/java/pcgen/persistence/lst/CampaignSourceEntry.java]: This message should be updated

File: code/src/java/pcgen/rules/persistence/CDOMKitLoader.java [hunk 2]
```diff
@@ -140,87 +127,6 @@ protected Kit getCDOMObject(LoadContext context, String name)
 	public void unloadLstFiles(LoadContext lc,
```
Comment (jdempsey-au) [inline code/src/java/pcgen/rules/persistence/CDOMKitLoader.java]: Is this method obsolete now? Should it be removed form the interface too?

File: code/src/java/pcgen/persistence/lst/CampaignSourceEntry.java (hunk 1 above)
Comment (thpr) [inline code/src/java/pcgen/persistence/lst/CampaignSourceEntry.java]: Both items are now completed.

**Pull request merged**